### PR TITLE
Add support for parquet key-value metadata

### PR DESCRIPTION
**Pull request by Maxxen**
This PR adds support for reading and writing parquet key-value metadata
- The parquet copy function now takes an optional `KV_METADATA: STRUCT` argument where every member represents a key-value pair. We cast all values to VARCHAR internally.
- There is a new `parquet_kv_metadata` table function that returns two `BLOB` columns for keys and values respectively.

### CODE REVIEW

**Comment by Tmonster** [comment]
R CMD check should pass if run again

File: extension/parquet/parquet_metadata.cpp [hunk 1]
```diff
@@ -429,13 +433,53 @@ void ParquetMetaDataOperatorData::LoadSchemaData(ClientContext &context, const v
...
+	auto reader = make_uniq<ParquetReader>(context, file_path, parquet_options);
+	idx_t count = 0;
+	DataChunk current_chunk;
+	current_chunk.Initialize(context, return_types);
+	auto meta_data = reader->GetFileMetadata();
+
+	for (idx_t col_idx = 0; col_idx < meta_data->key_value_metadata.size(); col_idx++) {
+		auto &entry = meta_data->key_value_metadata[col_idx];
+
+		current_chunk.SetValue(0, count, Value::BLOB_RAW(entry.key));
```
Comment (Mytherin) [inline extension/parquet/parquet_metadata.cpp:457]: Are both the key and the value blobs or are they limited to strings?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one minor comment

File: test/sql/copy/parquet/kv_metadata.test [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# name: test/sql/copy/parquet/kv_metadata.test
+# group: [parquet]
+
+require parquet
+
+statement ok
+COPY (SELECT 1, 'foo') TO  '__TEST_DIR__/kv_metadata_test.parquet' (FORMAT PARQUET, KV_METADATA {foo: 'bar', baz: 42});
+
+query II
+SELECT key::VARCHAR, value::VARCHAR FROM parquet_kv_metadata('__TEST_DIR__/kv_metadata_test.parquet');
```
Comment (Mytherin) [inline test/sql/copy/parquet/kv_metadata.test:10]: Perhaps also (1) try the function on a file without metadata, and (2) try the function with a glob/list of multiple Parquet files ('data/parquet/*.parquet')

File: extension/parquet/parquet_metadata.cpp (hunk 1 above)
Comment (Maxxen) [reply extension/parquet/parquet_metadata.cpp:457]: The standard says they are allowed to be anything, i.e. blobs, but from what I can tell, in practice its almost always utf8-encoded strings/json/text.

File: test/sql/copy/parquet/kv_metadata.test (hunk 2 above)
Comment (Maxxen) [reply test/sql/copy/parquet/kv_metadata.test:10]: Im just hooking into the existing parquet metadata reading code, so I assumed this all would work if it works for the normal metadata, but sure I can add some tests for KV specifically.

File: extension/parquet/parquet_metadata.cpp (hunk 1 above)
Comment (Mytherin) [reply extension/parquet/parquet_metadata.cpp:457]: If the standard says they can be anything then let's leave it as blobs ont he read-side. Perhaps on the write-side values could be either (1) cast to string, or (2) taking the literal bytes if it is of blob type? Keys always need to be utf-8 if we are using struct keys.

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**